### PR TITLE
Remove automapper from Implementation timescales

### PR DESCRIPTION
**Pull request by PhysicsJon**

### CODE REVIEW

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Solutions/Models/ImplementationTimescalesModel.cs [hunk 1]
```diff
@@ -1,11 +1,17 @@
-﻿using System;
-using NHSD.GPIT.BuyingCatalogue.EntityFramework.Catalogue.Models;
+﻿using NHSD.GPIT.BuyingCatalogue.EntityFramework.Catalogue.Models;
 
 namespace NHSD.GPIT.BuyingCatalogue.WebApp.Areas.Solutions.Models
 {
     public class ImplementationTimescalesModel : SolutionDisplayBaseModel
```
Comment (bjss-chris) [inline src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Solutions/Models/ImplementationTimescalesModel.cs:5]: Can we seal this now?

File: tests/NHSD.GPIT.BuyingCatalogue.WebApp.UnitTests/Areas/Solutions/Controllers/SolutionsControllerTests.cs [hunk 2]
```diff
@@ -774,11 +774,17 @@ public static async Task Get_HostingType_ValidSolutionForId_ReturnsExpectedViewR
...
+            var controller = new SolutionsController(
+                Mock.Of<IMapper>(),
+                mockService.Object,
+                Mock.Of<IMemoryCache>(),
+                Mock.Of<ISolutionsFilterService>(),
+                new FilterCacheKeySettings());
+
             await controller.Implementation(id);
 
             mockService.Verify(s => s.GetSolutionOverview(id));
```
Comment (bjss-chris) [inline tests/NHSD.GPIT.BuyingCatalogue.WebApp.UnitTests/Areas/Solutions/Controllers/SolutionsControllerTests.cs:790]: I think we can do something like this here.

```c#
public static async Task Get_ImplementationTimescales_ValidId_GetsSolutionFromService(
    [Frozen] CatalogueItemId id,
    [Frozen] Mock<ISolutionsService> solutionsServiceMock,
    SolutionsController controller)
{
    solutionsServiceMock.Setup(s => s.GetSolutionOverview(id)).ReturnsAsync((CatalogueItem)null);

    await controller.Implementation(id);

    solutionsServiceMock.Verify(s => s.GetSolutionOverview(id));
}
```

File: tests/NHSD.GPIT.BuyingCatalogue.WebApp.UnitTests/Areas/Solutions/Controllers/SolutionsControllerTests.cs [hunk 3]
```diff
@@ -802,50 +808,27 @@ public static async Task Get_HostingType_ValidSolutionForId_ReturnsExpectedViewR
...
                 new FilterCacheKeySettings());
 
-            var actual = (await controller.Implementation(id)).As<ViewResult>();
+            var actual = (await controller.Implementation(solution.Id)).As<ViewResult>();
 
             actual.Should().NotBeNull();
             actual.ViewName.Should().BeNullOrEmpty();
-            actual.Model.Should().Be(mockSolutionImplementationTimescalesModel);
+            actual.Model.Should().BeEquivalentTo(implementationTimescalesModel);
         }
```
Comment (bjss-chris) [inline tests/NHSD.GPIT.BuyingCatalogue.WebApp.UnitTests/Areas/Solutions/Controllers/SolutionsControllerTests.cs:832]: I think we can do this here.

```c#
public static async Task Get_ImplementationTimescales_ValidSolutionForId_ReturnsExpectedViewResult(
    [Frozen] CatalogueItem solution,
    [Frozen] Mock<ISolutionsService> mockService,
    SolutionsController controller)
{
    var implementationTimescalesModel = new ImplementationTimescalesModel(solution);

    mockService.Setup(s => s.GetSolutionOverview(solution.Id)).ReturnsAsync(solution);

    var actual = (await controller.Implementation(solution.Id)).As<ViewResult>();

    actual.Should().NotBeNull();
    actual.ViewName.Should().BeNullOrEmpty();
    actual.Model.Should().BeEquivalentTo(implementationTimescalesModel);
}
```

File: tests/NHSD.GPIT.BuyingCatalogue.WebApp.UnitTests/Areas/Solutions/Controllers/SolutionsControllerTests.cs (hunk 2 above)
Comment (PhysicsJon) [reply tests/NHSD.GPIT.BuyingCatalogue.WebApp.UnitTests/Areas/Solutions/Controllers/SolutionsControllerTests.cs:790]: I tried this and the test started to fail reliably. It's something to add to look at in the tech debt backlog

Comment (PhysicsJon) [reply tests/NHSD.GPIT.BuyingCatalogue.WebApp.UnitTests/Areas/Solutions/Controllers/SolutionsControllerTests.cs:790]: I hadn't tried the returns null bit, I'll see what happens

Comment (PhysicsJon) [reply tests/NHSD.GPIT.BuyingCatalogue.WebApp.UnitTests/Areas/Solutions/Controllers/SolutionsControllerTests.cs]: It works. I've added it to the latest commit

Comment (bjss-colin) [reply tests/NHSD.GPIT.BuyingCatalogue.WebApp.UnitTests/Areas/Solutions/Controllers/SolutionsControllerTests.cs]: Shouldn't need to apply `[Frozen]` to the `CatalogueItemId` or `CatalogueItem` in either test as it's typically added only to services/repositories/mockable types that are injected via the constructor to ensure that your test and system under test both work on the single instance.

**Review by bjss-colin — APPROVED** [review]
Left a comment with regards to `[Frozen]` usage but this is at your discretion.

**Review by bjss-chris — APPROVED** [review]
@bjss-colin is right about `[Frozen]` here (apologies, I should have noticed) but there's quite a bit of AutoFixture-related stuff to tidy up so can always be done as part of that.

**Pull request merged**